File: app/src/main/java/net/squanchy/speaker/SpeakerDetailsModule.java
package net.squanchy.speaker;

import net.squanchy.service.firebase.FirebaseAuthService;
import net.squanchy.service.repository.SpeakerRepository;

import dagger.Module;
import dagger.Provides;

@Module
class SpeakerDetailsModule {

    @Provides
    SpeakerDetailsService scheduleService(SpeakerRepository speakerRepository, FirebaseAuthService authService) {
        return new SpeakerDetailsService(speakerRepository, authService);
    }
}

File: app/src/main/java/net/squanchy/speaker/SpeakerDetailsModule.kt
package net.squanchy.speaker

import dagger.Module
import dagger.Provides
import net.squanchy.service.repository.AuthService
import net.squanchy.service.repository.SpeakerRepository

@Module
internal class SpeakerDetailsModule {

    @Provides
    fun scheduleService(speakerRepository: SpeakerRepository, authService: AuthService) =
        SpeakerDetailsService(speakerRepository, authService)
}
